chapters/classes.tex

```diff
@@ -281,37 +281,58 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
   function.
 \end{itemize}
 
+
+\subsection{Component Variability Prefixes}\label{component-variability-prefixes-discrete-parameter-constant}
+
+The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration are called \firstuse[variability!prefix]{variability prefixes} and are the basis for defining in which situation the variable values of a component are initialized (see \cref{events-and-synchronization} and \cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed during \firstuse{transient analysis} (i.e., solution of initial value problem of the hybrid DAE).
```

Comment:
This is the first use of the term _transient analysis_, but introducing the definition here looks odd to me.  The question is whether we'd really want to move the introduction somewhere else, considering that the term is only used in this chapter?

One alternative to consider would be to define both _initialization problem_ and _transient analysis_ after the definition of _flattening_ in the introduction: https://specification.modelica.org/master/introduction1.html#some-definitions

Reply:
I do not implement initialization or simulation, so to me the distinction between "solution of initial value problem of the hybrid DAE" and initialization is not clear. I always thought they were the same. The "transient analysis" term is only used here, but it is a crucial concept in how/when to determine parameters. So if it is not the same as initialization, I would like this difference explained better. Other term that I perhaps confuse is flattening vs translation. We have the official definition of flattening (the translation of ...) but there are many places where we call it the translation phase and not the flattening phase. Are they the same or different? Do we have an overview of all the steps or at least the suggestion of how a model gets from Modelica code to a simulation, where we can clearly see how all the steps like flattening, translation, clock partitioning, initialization, transient analysis, simulation, etc are related?

Reply:
Sounds like a good idea, but I think we have to start thinking seriously about the scope creep in this PR.  How about opening a separate issue/PR about this, and revert the introduction of `\firstuse` here to avoid merge conflicts?

Reply:
> I do not implement initialization or simulation, so to me the distinction between "solution of initial value problem of the hybrid DAE" and initialization is not clear. I always thought they were the same. The "transient analysis" term is only used here, but it is a crucial concept in how/when to determine parameters. 

The "transient analysis" is the phase _after_ the initialization; where you have "transients" (i.e. time-varying behavior). That's a fairly normal definition for circuit simulations (and is often contrasted with static simulations). I don't see that there's a clear separation between "transient analysis" and "simulation" for us.

The problem here seems to be:
"and when they are changed in transient analysis (= solution of initial value problem of the hybrid DAE):"
That was added in Modelica 1.2, extended slightly in 1.3 and has been basically unchanged since then.

Here 'solution of initial value problem' isn't related to 'initialization' but to simulation from a given initial value (the initial value problem cp. boundary value problem).

So there is no actual error, but it is not helpful as it is introducing one term in terms of another that isn't defined (and 'initial value problem' is used in two other places  - as far as I can see only 50% of them are correct).

I will make a PR clearing up this.

Reply:
The new PR is #3146 (and clarified previous comment).

Reply:
@eshmoylova: OK to put faith in #3146 and remove the introduced use of `\firstuse` from the present PR?

Reply:
OK, resolving, given that it would be merged/updated after #3146.

Reply:
Good.  The introduction of `\firstuse` has been reverted.